### PR TITLE
fix(api): ensure memtable schema and columns match

### PR DESCRIPTION
**Pull request by deepyaman**
## Description of changes

Before this change, if you pass a `SchemaLike` that isn't actually a `Schema`, it won't automatically rename the auto-generated pandas column names. For example:

```python
import pandas as pd

df = pd.DataFrame([[1, 2], [3, 4]])
schema = {"a": "int64", "b": "int64"}
t = ibis.memtable(df, schema=schema)
t.a.execute()  # Raises a KeyError
```

This PR goes ahead and constructs the `Schema`, if passed—work that was going to be done anyway. That way, `getattr(schema, "names")` finds the passed names, and the rename is performed.

### CODE REVIEW

**Review by cpcloud — CHANGES_REQUESTED** [review]
You need to monkey patch the default backend setting

File: ibis/expr/tests/test_api.py [hunk 1]
```diff
@@ -143,6 +143,15 @@ def test_duplicate_columns_in_memtable_not_allowed():
         ibis.memtable(df)
 
 
+def test_memtable_column_names_match_schema():
+    pd = pytest.importorskip("pandas")
+
+    df = pd.DataFrame([[1, 2], [3, 4]])
+    schema = {"a": "int64", "b": "int64"}
+    t = ibis.memtable(df, schema=schema)
+    t.a.execute()  # Raises a KeyError if the column name does not match
```
Comment (cpcloud) [inline ibis/expr/tests/test_api.py:152]: Execution tests should not be under this directory

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: ibis/expr/tests/test_api.py (hunk 1 above)
Comment (deepyaman) [reply ibis/expr/tests/test_api.py:152]: How about I don't execute, and instead I check that the columns of the dataframe underlying the memtable have been renamed/match--would that work?

**Comment by deepyaman** [comment]
> You need to monkey patch the default backend setting

Where?

**Comment by cpcloud** [comment]
> > You need to monkey patch the default backend setting
> 
> Where?
> 
> (Also, why? Did I cause the nix failure; doesnt seem related to my change, even though its related to the default backend?)

The default backend attribute is global state, and tests run in a random order. This means that **any** of our suites could have failed in the way, you just got lucky 😄

File: ibis/expr/tests/test_api.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/tests/test_api.py:152]: That seems better, and then you don't need to muck around with using `monkeypatch` to set the default backend.

Comment (deepyaman) [reply ibis/expr/tests/test_api.py]: Done!

**Review by cpcloud — APPROVED** [review]

**Pull request merged**